src/jdk.incubator.vector/share/classes/jdk/incubator/vector/AbstractMask.java

```diff
@@ -138,7 +140,7 @@ public <F> VectorMask<F> cast(VectorSpecies<F> dsp) {
                 this.getClass(), vspecies().elementType(), vspecies().laneCount,
                 species.maskType(), species.elementType(), vspecies().laneCount,
                 this, species,
-                (m, s) -> s.maskFactory(m.getBits()).check(s));
+                (m, s) -> VectorMask.fromLong(s, m.toLong()).check(s));
```

Comment:
Any issues with the original `maskFactory()` ?

Reply:
We cannot typecast value object with another value type, thus directly passing payload (m.getBits()) to maskFactory will cause compilation failures.

Reply:
Sounds reasonable! But I'm curious why we didn't meet this issue before?